bgpd/bgp_route.c

```diff
@@ -7439,6 +7653,26 @@ route_vty_out_detail (struct vty *vty, struct bgp *bgp, struct prefix *p,
       if (binfo->extra && binfo->extra->damp_info)
 	bgp_damp_info_vty (vty, binfo, json_path);
 
+      if (bgp_labeled_safi(safi) && binfo->extra)
+        {
+          uint32_t label = label_pton(binfo->extra->tag);
+          if (json_paths)
+            json_object_int_add(json_path, "remote-label", label);
+          else
+            vty_out(vty, "      Remote label: %d", label);
+        }
+
+      if (CHECK_FLAG (attr->flag, ATTR_FLAG_BIT (BGP_ATTR_LABEL_INDEX)))
+        {
+          if (json_paths)
+            json_object_int_add(json_path, "label-index", attr->extra->label_index);
+          else
+            vty_out(vty, ", Label Index: %d", attr->extra->label_index);
```

Comment:
Please take a look at the output below:
```
# show bgp ipv4 labeled-unicast 2.2.2.2/32
BGP routing table entry for 2.2.2.2/32
Local label: 3
Paths: (1 available, best #1)
  Advertised to non peer-group peers:
                                       10.0.1.1 10.0.2.3 10.0.2.4
  Local
    0.0.0.0 from 0.0.0.0 (10.0.2.2)
      Origin IGP, metric 0, localpref 100, weight 32768, valid, sourced, local, best
, Label Index: 51
      AddPath ID: RX 0, TX 3
      Last update: Fri Apr  7 22:11:58 2017
```
This lack of proper padding happens when we have a "Label Index" to show but not a "Remote label".

Also, it would be nice to update the "Line N" comments in this function as they are outdated now.